GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -34,12 +34,30 @@ public GVFSVerb(bool validateOrigin = true)
         public abstract string EnlistmentRootPathParameter { get; set; }
 
         [Option(
-            GVFSConstants.VerbParameters.Mount.ServiceName,
-            Default = GVFSConstants.Service.ServiceName,
+            GVFSConstants.VerbParameters.Mount.InternalUseOnly,
             Required = false,
             HelpText = "This parameter is reserved for internal use.")]
+        public string InternalParameters
+        {
+            set
+            {
+                if (!string.IsNullOrEmpty(value))
+                {
+                    MountInternal mountInternal = MountInternal.FromJson(value);
+                    if (!string.IsNullOrEmpty(mountInternal.ServiceName))
+                    {
+                        this.ServiceName = mountInternal.ServiceName;
+                    }
+
+                    this.StartedByService = mountInternal.StartedByService;
+                }
+            }
+        }
+
         public string ServiceName { get; set; }
 
+        public bool StartedByService { get; set; }

```

Comment:
Minor: Can `ServiceName` and `StartedByService` be changed to `protected`?

(It actually looks like `Unattended`, `ServicePipeName`, and `Output` can probably be `protected` as well)

Reply:
They can be, but then so could everything else in these classes. This is the top of the stack so there isn't any other code calling these methods/properties other than the verbs themselves.